content/docs/io/overview.md

```diff
@@ -6,7 +6,109 @@ menu:
     parent: io
 ---
 
-This page has not been worked on yet. If you'd like to contribute visit the [doc-push]
-repo.
+Rust [`std::io`] module provides traits that synchronously access resources like
+files and sockets.  These `Read` and `Write` traits are implemented in other
+modules including [`std::fs`] and `std::net`.
 
-[doc-push]: https://github.com/tokio-rs/doc-push
+The [`tokio`] crate comes with a variation on those types, [`AsyncRead`] and
+[`AsyncWrite`].  The primary difference between [`AsyncRead`] and `std::io::Read`
+is the [`futures::Async`] return values.  The implementation returns an `Error`,
+`NotReady`, or `Ready` value.
+
+## Non Blocking
```

Comment:
We might want to name this section "Blocking vs. Non-Blocking" or something like that.

You can also look at [Node's page](https://nodejs.org/en/docs/guides/blocking-vs-non-blocking/) for inspiration.